File: typescript/tests/player.test.ts
import { describe, expect, test } from 'bun:test';

import chalk from 'chalk';

import { Player } from '../src/player.ts';
import { PlayerSettings } from '../src/settings.ts';

describe('player', () => {
    test('new black player', () => {
        const player = Player.black(PlayerSettings.default());
        expect(player.human()).toBe(true);
        expect(player.computer()).toBe(false);
        expect(player.canPlay).toBe(true);
        expect(player.toString()).toBe(`${chalk.magenta('BLACK')} | Human    | Moves: 0`);
    });

    test('new white player', () => {
        const player = Player.white(PlayerSettings.default());
        expect(player.human()).toBe(true);
        expect(player.computer()).toBe(false);
        expect(player.canPlay).toBe(true);
        expect(player.toString()).toBe(`${chalk.cyan('WHITE')} | Human    | Moves: 0`);
    });

    test('set human and computer', () => {
        const player = Player.black(PlayerSettings.default());
        expect(player.human()).toBe(true);

        player.setComputer();
        expect(player.computer()).toBe(true);
        expect(player.human()).toBe(false);

        player.setHuman();
        expect(player.human()).toBe(true);
        expect(player.computer()).toBe(false);
    });

    test('player type string', () => {
        const player = Player.black(PlayerSettings.default());
        expect(player.typeString()).toBe('Human   ');

        player.setComputer();
        expect(player.typeString()).toBe('Computer');
    });

    test('reset player', () => {
        const player = Player.white(PlayerSettings.default());
        player.canPlay = false;
        player.reset();
        expect(player.canPlay).toBe(true);
        expect(player.toString()).toBe(`${chalk.cyan('WHITE')} | Human    | Moves: 0`);
    });
});

File: kotlin-native/src/nativeTest/kotlin/othello/PlayerTest.kt
package othello

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertTrue

class PlayerTest {
    @Test
    fun newPlayer() {
        val player = Player.default()
        assertEquals(Disk.Black, player.disk)
        assertTrue(player.human())
        assertTrue(player.canPlay)
        assertEquals(0, player.roundsPlayed)
        assertEquals(PlayerSettings.default(), player.settings)
    }

    @Test
    fun resetPlayer() {
        val player = Player(Disk.White, PlayerSettings.default())
        player.canPlay = false
        player.roundsPlayed = 10
        player.reset()
        assertTrue(player.canPlay)
        assertEquals(0, player.roundsPlayed)
    }

    @Test
    fun setHuman() {
        val player = Player.default()
        player.setComputer()
        assertTrue(player.computer())
        assertEquals(PlayerType.Computer, player.playerType)
        player.setHuman()
        assertTrue(player.human())
        assertEquals(PlayerType.Human, player.playerType)
    }

    @Test
    fun playerTypeString() {
        val player = Player.default()
        assertEquals("Human   ", player.typeString())

        player.setComputer()
        assertEquals("Computer", player.typeString())
    }
}
